dascore/io/index/backend.py

```
@@ -838,7 +847,38 @@ def _flatten(self, df: pd.DataFrame, attr_meta: pd.DataFrame) -> pd.DataFrame:
+        # chunking); warn once per backend per colliding name set so users
+        # learn about the shadowing without a warning on every access.
+        already_warned = getattr(self, "_warned_attr_clobber", set())
+        if clobbered and clobbered not in already_warned:
+            self._warned_attr_clobber = already_warned | {clobbered}
+            names = ", ".join(sorted(clobbered))
+            msg = (
+                f"Attr(s) {names} collide with coordinate envelope columns "
+                "and are omitted from the flat contents; query them via the "
+                "_attrs namespace."
+            )
+            warnings.warn(msg, UserWarning)
```

Ruff (0.15.21)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)